GVFS/GVFS.Notifications/VFSForGit.Mac/StatusMenuItem/IPC/VFSMessageParser.m

```diff
@@ -4,20 +4,23 @@
 
 @implementation VFSMessageParser
 
-+ (BOOL)tryParseData:(NSData *_Nonnull)data
-             message:(NSDictionary *_Nonnull __autoreleasing *_Nonnull)parsedMessage
-               error:(NSString *_Nonnull __autoreleasing *_Nonnull)error
++ (BOOL)tryParseData:(NSData *)data
+             message:(NSDictionary *__autoreleasing *)parsedMessage
+               error:(NSError *__autoreleasing *)error
 {
     NSParameterAssert(parsedMessage);
     
     NSString *messageStr;
     if (!(messageStr = [[NSString alloc] initWithData:data
                                              encoding:NSUTF8StringEncoding]))
     {
-        if (error)
+        if (error != nil)
         {
-            *error = [NSString stringWithFormat:@"%@: error reading data",
-                      NSStringFromSelector(_cmd)];
+            NSString *info = [NSString stringWithFormat:@"%@: ERROR: error reading data.",
+                              NSStringFromSelector(_cmd)];
+            *error = [NSError errorWithDomain:@"VFS For Git"
```

Comment:
Move the domain string out to a single extern constant (declaration in some common header, single initialization in a .m file). Also see naming convention for custom error domains: https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ErrorHandlingCocoa/ErrorObjectsDomains/ErrorObjectsDomains.html
_You can create your own error domains and error codes for use in your own frameworks, or even in your own applications. It is recommended that the string constant for the domain be of the form com.company.framework_or_app.ErrorDomain_.

Reply:
Moved Notification Agent specific errors to a separate file - `VFSNotificationErrors.h`. `VFSNotificationErrors.m` has only a single line with definition of the Domain - `VFSForGitNotificationErrorDomain`.